Mammoth/TSE/AIManeuvers.cpp

```diff
@@ -1401,9 +1402,127 @@ void CAIBehaviorCtx::ImplementFireWeaponOnTarget (CShip *pShip,
 	//	Select the appropriate weapon. If we're not given a weapon, then choose the
 	//	best one.
 
-	DeviceNames iWeaponToFire;
-	Metric rWeaponRange;
-	if (iWeapon == -1)
+	std::vector<CInstalledDevice*> pWeaponsToFire;
```

Comment:
Instead of changing ImplementFireWeaponOnTarget, would it be possible to create ImplementFireAllWeaponsOnTarget (or something like that) and call it only when UsesAllPrimaryWeapons is true?